torch/accelerator/memory.py

```diff
@@ -0,0 +1,204 @@
+import collections
+from typing import Any
+
+import torch
+
+from ._utils import _device_t, _get_device_index
+
+
+__all__ = [
+    "empty_cache",
+    "max_memory_allocated",
+    "max_memory_reserved",
+    "memory_allocated",
+    "memory_reserved",
+    "memory_stats",
+    "memory_stats_as_nested_dict",
+    "reset_accumulated_memory_stats",
+    "reset_peak_memory_stats",
+]
+
+
+def empty_cache() -> None:
+    r"""Release all unoccupied cached memory currently held by the caching
+    allocator so that those can be used in other application.
+
+    .. note:: This function is a no-op if the memory allocator for the current
+        :ref:`accelerator <accelerators>` has not been initialized.
+    """
+    if not torch._C._accelerator_isAllocatorInitialized():
+        return
+    torch._C._accelerator_emptyCache()
+
+
+def memory_stats_as_nested_dict(device: _device_t = None, /) -> dict[str, Any]:
+    r"""Return the result of :func:`~torch.accelerator.memory_stats` as a nested dictionary.
+
+    Args:
+        device (:class:`torch.device`, str, int, optional): a given device that must match the current
+            :ref:`accelerator<accelerators>` device type. If not given,
+            use :func:`torch.accelerator.current_device_index` by default.
+    """
+    if not torch._C._accelerator_isAllocatorInitialized():
+        return {}
+    device = _get_device_index(device, optional=True)
+    return torch._C._accelerator_getDeviceStats(device)
+
+
+def memory_stats(device: _device_t = None, /) -> dict[str, Any]:
```

Comment:
Could we use device_index for all these function for the documentation name?
I think passing an int is the expected way. And passing a torch.device is a special case.

Reply:
Make sense to change to device index.
We can keep the positional-only slash (`/`) as discussed in [#132204](https://github.com/pytorch/pytorch/pull/132204#discussion_r1806724873), or removing it is also acceptable. Plenty of libraries use either style. Let me know if you'd like to lean one way for consistency.